warp10/src/main/java/io/warp10/continuum/gts/MetadataTextComparator.java

```diff
@@ -19,163 +19,237 @@
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.Comparator;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 
 import io.warp10.continuum.store.thrift.data.Metadata;
 
 /**
- * Sort Metadata according to text order (on name then on labels)
- * 
+ * Sort Metadata according to field values (labels, attributes or classname if null provided)
  */
 public class MetadataTextComparator implements Comparator<Metadata> {
 
   private final List<String> fields;
-  
+  private final boolean considerAttributes;
```

Comment:
`considerAttributes` is not taken into account when there `fields` is `null` or empty.
This is totally OK because I don't see why metadata ordering would leave order based on attributes unspecified.

However, for clarity I would add a comment for this field clearly stating that. Maybe renaming the variable `fieldsConsiderAttributes` would made it even clearer.

Reply:
Ok, renamed and commented